flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/BackPressureSampleService.java

```diff
@@ -53,64 +53,58 @@
 
 		this.numSamples = numSamples;
 		this.delayBetweenSamples = checkNotNull(delayBetweenSamples);
-		this.scheduledExecutor = checkNotNull(scheduledExecutor, "The scheduledExecutor must not be null.");
+		this.scheduledExecutor = checkNotNull(scheduledExecutor);
 	}
 
 	/**
-	 * Returns a future that completes with the back pressure ratio of a task.
+	 * Schedules to sample the task back pressure and returns a future that completes
+	 * with the back pressure ratio.
 	 *
 	 * @param task The task to be sampled.
-	 * @return A future of the task back pressure ratio.
+	 * @return A future containing the task back pressure ratio.
 	 */
 	public CompletableFuture<Double> sampleTaskBackPressure(BackPressureSampleableTask task) {
+		if (!task.isRunning()) {
+			throw new IllegalStateException("Cannot sample task. Because it is not running.");
```

Comment:
It is better to give some debug info here. E.g we could give `task.toString()` in the message, then in the specific task implementation, it can provide `executionId` info as before.